app/controllers/users_controller.rb

```diff
@@ -1,9 +1,19 @@
 # frozen_string_literal: true
 
 class UsersController < WithUserController
-  prepend_before_action { set_user(:id) }
-
   def show
     render json: @user
   end
+
+  def destroy
+    @user.user_counters.destroy_all
+    @user.favorite_tools.destroy_all
+    render json: "", status: 204
+  end
```

Comment:
does this also delete the user object?

Reply:
Oops! Forgot the most important thing. I'll add that in.